### PR TITLE
3scale toolbox import openapi

### PR DESCRIPTION
**Pull request by rafamqrs**
# Changes
Add new task which runs a 3scale toolbox to import service.

# Submitter Checklist
These are the criteria that every PR should meet, please check them off as you
review them:

 Follows the authoring recommendations

- [X]  Includes docs (if user facing)
- [X]  Includes tests (if functionality of task changed or new task added)
- [X]  Commit messages follow commit message best practices
- [X]  Complies with Catalog Organization TEP, see example. Note An issue has been filed to automate this validation
- [X]  File path follows <kind>/<name>/<version>/name.yaml

 

    - [X] Has README.md at <kind>/<name>/<version>/README.md
    - [X]  Has mandatory metadata.labels - app.kubernetes.io/version the same as the <version> of the resource
    - [X]  Has mandatory metadata.annotations tekton.dev/pipelines.minVersion
    - [X]  mandatory spec.description follows the convention

  ```

  spec:
    description: >-
      one line summary of the resource

      Paragraph(s) to describe the resource.
  ```
See the contribution guide
for more details.

### CODE REVIEW

File: task/3scale-toolbox-import/0.1/tests/run.yaml
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/tests/run.yaml:48]: This image might not be public. Any specific reason for choosing this image in place of the one from gcr?

File: task/3scale-toolbox-import/0.1/README.md
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/README.md:8]:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
```

File: task/3scale-toolbox-import/0.1/README.md
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/README.md:1]: Can you please replace this name with the name you want for this task?

File: task/3scale-toolbox-import/0.1/OWNERS
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/OWNERS:3]: are you willing to add yourself as the owner of this Task? If so, can you please add your GitHub handle over here?

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml:7]:
```suggestion
    tekton.dev/platforms: 'linux/amd64,linux/s390x,linux/ppc64le,linux/arm64'
```

File: task/3scale-toolbox-import/0.1/README.md
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/README.md:37]: in the task it's mentioned several other platforms as well so can you please update here as well?

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml:42]: `$(inputs.params.DESTINATION)` is in older notation, can you please this with `$(params.DESTINATION)`? Also in subsequent lines?

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml:59]: instead of setting this via volumes... can this also be mounted via workspaces?

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml:55]:
```suggestion
      workingDir: /workspace/output/$(inputs.params.CONTEXT_DIR)
```

**Review by vinamra28 — CHANGES_REQUESTED** [review]
hey @rafamqrs. I have left a couple of comments

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml]: Remove, it can be mounted via workspace, but I think the best option is to use the URL.

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml]: Updated

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml]: Updated, now i'm using the $(params.*)

File: task/3scale-toolbox-import/0.1/README.md
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/README.md]: Done

File: task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/3scale-toolbox-import.yaml]: Removed the double "l"

File: task/3scale-toolbox-import/0.1/OWNERS
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/OWNERS]: Added my users as an owner

File: task/3scale-toolbox-import/0.1/README.md
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/README.md]: Done

File: task/3scale-toolbox-import/0.1/README.md
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/README.md]: Remove the duplicated .yaml

File: task/3scale-toolbox-import/0.1/tests/run.yaml
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/tests/run.yaml]: Removed this image and the task git-clone

File: task/3scale-toolbox-import/0.1/OWNERS
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/OWNERS:3]:
```suggestion
approvers:
- rafamqrs
```

Comment (vinamra28) [reply task/3scale-toolbox-import/0.1/OWNERS:3]: also you need to make a PR to tektoncd/community to add yourself in the org in order to be an owner for this resource. You can take reference of the following PR https://github.com/tektoncd/community/pull/538

File: task/3scale-toolbox-import/0.1/tests/run.yaml
Comment (vinamra28) [inline task/3scale-toolbox-import/0.1/tests/run.yaml:19]: is it a short lived cluster or will it be available all the time? Asking because once the PR is merged the task will be tested on a nightly basis and if the cluster goes down then it will result in the task's test failure.

Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/tests/run.yaml:19]: it is a short liver cluster, but it will be available until  11/20/21 01:20 UTC. The only that I test this integration is using a 3scale instance running, without it, I can't test.

File: task/3scale-toolbox-import/0.1/OWNERS
Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/OWNERS:3]: done, I've created a PR to tektoncd/community 

Now, I'm waiting for the PR approval: https://github.com/tektoncd/community/pull/563

File: task/3scale-toolbox-import/0.1/tests/run.yaml
Comment (vinamra28) [reply task/3scale-toolbox-import/0.1/tests/run.yaml:19]: how about you mock the call? You can take the example of [github-add-comment task tests](https://github.com/tektoncd/catalog/tree/main/task/github-add-comment/0.5/tests)

Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/tests/run.yaml:19]: It is a little bit hard to mock the endpoint, let me try to explain:
I'm using the 3scale toolbox cli from an image, it was built using the ruby, and basically it abstract all the interaction with the 3scale API, so to mock the calls, I will have to open the 3scale toolbox code and identify all request to 3scale API, due to there are many endpoints the toolbox called before saving the API. 

I can keep the cluster running maybe for one week if necessary, but if you think is better mock the call, I think is best to use the 3scale api instead of the 3scale toolbox.

Comment (rafamqrs) [reply task/3scale-toolbox-import/0.1/tests/run.yaml:19]: Hi @vinamra28, do you agree if I continue using the cluster?

**Comment by rafamqrs** [comment]
@vinamra28 Do you have any news? I've noticed that the pull request to add me as the owner is still open. If you have any suggestions, please let me know, I can update or close this pull.

**Comment by vinamra28** [comment]
> @vinamra28 Do you have any news? I've noticed that the pull request to add me as the owner is still open. If you have any suggestions, please let me know, I can update or close this pull.

apologies for late replies :pray: . I'll try to get the PR on tektoncd/community in and then we can move ahead with this PR.

**Comment by vinamra28** [comment]
@rafamqrs tektoncd/community#563 is merged. Probably you can proceed with resolving other comments

**Comment by rafamqrs** [comment]
> @rafamqrs [tektoncd/community#563](https://github.com/tektoncd/community/pull/563) is merged. Probably you can proceed with resolving other comments

@vinamra28 I resolved all the issues, but I'm wondering about the cluster of 3scale, my cluster was deleted, so Do I need to create one and change the URL before we can proceed?

**Comment by vinamra28** [comment]
@rafamqrs instead of creating one, can we mock the request? Probably you can take the example of github-add-comment task. There we have mocked the request instead of directly calling the actual API
Also did you get an invite to join the tektoncd org?

**Comment by rafamqrs** [comment]
> @rafamqrs instead of creating one, can we mock the request? Probably you can take the example of github-add-comment task. There we have mocked the request instead of directly calling the actual API Also did you get an invite to join the tektoncd org?

Unfortunately we can't mock, the 3scale toolbox is client built in Ruby that abstract all the request to the 3scale API, so it is simple function from toolbox to call different endpoints and import your API. If I need to mock all the endpoints is better to using the 3scale API directly because I wouldn't need to open the toolbox code and understand all the request. 

No, I didn't get an invite, Could you send the invite?

**Pull request closed without merging**
